res/controllers/Pioneer-DDJ-SB3-scripts.js

```diff
@@ -1412,6 +1411,16 @@ PioneerDDJSB3.EffectUnit = function(unitNumber) {
         },
     });
 
+    this.shiftKnob = new components.Pot({
+        unshift: function() {
+            this.input = function(channel, control, value) {
+                value = (this.MSB << 7) + value;
+
+                engine.setValue(eu.group, "super1", value / this.max);
+            };
+        },
```

Comment:
This is sufficient, the rest is already implement in the `components.Pot`
```suggestion
        inKey: "super1",
        group: eu.group
```